cmd/juju/action/cancel.go

```diff
@@ -87,14 +87,15 @@ func (c *cancelCommand) Run(ctx *cmd.Context) error {
 		return errors.Errorf("identifier(s) %q matched action(s) %q, but no actions were canceled", c.requestedIds, actionTags)
 	}
 
-	var unCanceledActions []string
+	var unCanceledActions []names.ActionTag
 	var canceledActions []params.ActionResult
-	for _, result := range actions.Results {
-		if result.Error != nil {
-			unCanceledActions = append(unCanceledActions, result.Action.Tag)
-			continue
+
+	for i, result := range actions.Results {
+		if result.Action != nil {
```

Comment:
are you saying that we are returning 'nil' for the Action, but not Error? Since it appears that we were checking for Error as a way to identify the action wasn't cancelled.

It feels like we actually want 3 states here. I'm not sure if it is worthwhile, but something to consider:

1) Action as requested was cancelled
2) Action was not cancelled because it had already completed
3) Action was not cancelled because you don't have rights to cancel it.

I'm thinking we may want 3 because:
1) I did what you asked me to do
2) You may need to deal with the fact that you were too late. Either revert whatever the Action did, or not worry because you only cancelled because it was taking too long.
3) You may need to find someone else to help you cancel this if you really need it stopped.

It feels like from a user perspective, what they need to do for (2) is different than (3) thus clearly identifying it to them would be helpful.

Do you agree? Is it worth the complexity?

The fact that we're not tracking and reporting the Error means that we aren't giving users Visibility into why the action isn't canceled, only *which* actions weren't cancelled. And that seems bad, to me.